### PR TITLE
feat(BarChart): add hideBarLabel prop which hides label of each bar

### PR DESCRIPTION
**Pull request by DmitriyT51**
[PRI-10](https://toptal-core.atlassian.net/browse/PRI-10)

### Description

- Add optional `hideBarLabel ` prop to `BarChart` component which allows hiding label of each bar
- Add story about how to use `hideBarLabel ` prop
- Add jest&cypress tests
- Add changeset for `Picasso-Charts` package

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Without prop (false by default) | With passed prop |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

File: packages/picasso-charts/src/BarChart/BarChart.tsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ export interface Props<K extends string | number | symbol>
   testIds?: {
     tooltip?: string
   }
+  barLabelHidden?: boolean
```
Comment (ozgurkececioglu) [inline packages/picasso-charts/src/BarChart/BarChart.tsx:48]: I would prefer `hideBarLabel` to be more convenient. WDYT?

File: packages/picasso-charts/src/BarChart/story/index.jsx [hunk 2]
```diff
@@ -105,3 +111,8 @@ page
     },
     delay: 500
   })
+  .addExample('BarChart/story/BarLabelHidden.example.tsx', {
+    title: 'Hidden bar label',
+    description: 'You can hide label of each bar via `barLabelHidden` prop.',
+    delay: 500
+  })
```
Comment (ozgurkececioglu) [inline packages/picasso-charts/src/BarChart/story/index.jsx:118]: I would suggest skipping visuals for jest and using happoSnapshot in cypress side

```suggestion
  }) // picasso-skip-visuals
```

File: .changeset/seven-terms-sniff.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso-charts': minor
+---
+
+Added new optional prop for BarChart component which allows hiding labels of all bars
```
Comment (TomasSlama) [inline .changeset/seven-terms-sniff.md:5]:
```suggestion
---
'@toptal/picasso-charts': minor
---

---
### BarChart

- Add new optional prop `barLabelHidden` for BarChart component which allows hiding labels of all bars.
```

We should follow [changeset guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md)

Comment (TomasSlama) [reply .changeset/seven-terms-sniff.md:5]: We should follow [changeset guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md)

File: packages/picasso-charts/src/BarChart/BarChart.tsx (hunk 1 above)
Comment (TomasSlama) [reply packages/picasso-charts/src/BarChart/BarChart.tsx:48]: Also, we should provide a comment above the prop, it is used for generating docs in storybook

File: packages/picasso-charts/src/BarChart/BarChart.tsx [hunk 4]
```diff
@@ -89,6 +90,7 @@ const BarChart = <K extends string>({
   labelKey,
   getBarLabelColor = defaultGetBarLabelColor,
   testIds,
+  barLabelHidden = false,
```
Comment (TomasSlama) [inline packages/picasso-charts/src/BarChart/BarChart.tsx:93]: I think, this default value should be in `defaultProps`

File: cypress/component/BarChart.spec.tsx [hunk 5]
```diff
@@ -146,4 +146,12 @@ describe('BarChart', () => {
     hoverOverBar('Los-Angeles')
     assertCustomTooltipContent('Infected: 2780Recovered: 3908')
   })
+
+  it('hides label of each bar via passed `hideBarLabel` prop', () => {
+    mount(<TestBarChart hideBarLabel={true} />)
```
Comment (TomasSlama) [inline cypress/component/BarChart.spec.tsx:151]:
```suggestion
    mount(<TestBarChart hideBarLabel />)
```

File: packages/picasso-charts/src/BarChart/BarChart.tsx [hunk 6]
```diff
@@ -155,10 +158,12 @@ const BarChart = <K extends string>({
               key={dataKey}
               dataKey={dataKey}
               label={
-                <BarChartLabel
-                  dataKey={dataKey}
-                  getBarLabelColor={getBarLabelColor}
-                />
+                !hideBarLabel && (
```
Comment (TomasSlama) [inline packages/picasso-charts/src/BarChart/BarChart.tsx:161]: I think this double negation looks weird 🤔, what do you think, if we use it like `hideBarLabel ? null : (<BarChartLabel ... />)`

File: packages/picasso-charts/src/BarChart/story/index.jsx [hunk 7]
```diff
@@ -105,3 +111,8 @@ page
     },
     delay: 500
   })
+  .addExample('BarChart/story/HideBarLabel.example.tsx', {
+    title: 'Hide bar label',
+    description: 'You can hide label of each bar via `hideBarLabel` prop.',
+    delay: 500
+  }) // picasso-skip-visuals
```
Comment (TomasSlama) [inline packages/picasso-charts/src/BarChart/story/index.jsx:118]: 👍, please don't forget to also remove the already committed snapshot [barchart-hidden-bar-label-snap.png](https://github.com/toptal/picasso/pull/2575/files#diff-4180fceed78d4bb83be41f1c11a6c4717ac49eaa167b60ff05dd940cd0e08cb9)

File: packages/picasso-charts/src/BarChart/BarChart.tsx (hunk 6 above)
Comment (DmitriyT51) [reply packages/picasso-charts/src/BarChart/BarChart.tsx:161]: `null` is not acceptable for `Bar` `label` prop. I can replace `null` with either `false` or `undefined`. What do you think?

Comment (TomasSlama) [reply packages/picasso-charts/src/BarChart/BarChart.tsx:161]: `undefined` is probably better 👍

**Review by TomasSlama — APPROVED** [review]
good job 👍

File: cypress/component/BarChart.spec.tsx [hunk 8]
```diff
@@ -146,4 +146,14 @@ describe('BarChart', () => {
     hoverOverBar('Los-Angeles')
     assertCustomTooltipContent('Infected: 2780Recovered: 3908')
   })
+
+  it('hides label of each bar via passed `hideBarLabel` prop', () => {
+    mount(<TestBarChart hideBarLabel />)
+
+    // Wait for bars from BarChart to finish height animation
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(1000)
```
Comment (ozgurkececioglu) [inline cypress/component/BarChart.spec.tsx:155]: I also searched for this and tried many things, but I couldn't find a good solution. So, let's continue with this 👍

**Review by ozgurkececioglu — APPROVED** [review]
I have two more comments, other than that, good job 👍

**Review by rodrigofd — APPROVED** [review]
Looks good! LGTM

File: packages/picasso-charts/src/BarChart/BarChart.tsx [hunk 9]
```diff
@@ -45,6 +45,8 @@ export interface Props<K extends string | number | symbol>
   testIds?: {
     tooltip?: string
   }
+  /** Hides label of each bar */
+  hideBarLabel?: boolean
```
Comment (denieler) [inline packages/picasso-charts/src/BarChart/BarChart.tsx:49]: it's better probably to have `showBarLabel` prop instead (https://github.com/ryanmcdermott/clean-code-javascript#avoid-negative-conditionals)

File: packages/picasso-charts/src/BarChart/story/index.jsx [hunk 10]
```diff
@@ -105,3 +111,8 @@ page
     },
     delay: 500
   })
+  .addExample('BarChart/story/HideBarLabel.example.tsx', {
+    title: 'Hide bar label',
+    description: 'You can hide label of each bar via `hideBarLabel` prop.',
+    delay: 500
```
Comment (denieler) [inline packages/picasso-charts/src/BarChart/story/index.jsx:117]: this is not needed if you skip visual test for this example (and replace it with Cypress test)

Comment (DmitriyT51) [reply packages/picasso-charts/src/BarChart/story/index.jsx:117]: Do you mean that I need to get rid of `HideBarLabel.example.tsx` and it's corresponding usage here in `story/index.jsx` ?
It was done to show this new functionality in storybook

Comment (denieler) [reply packages/picasso-charts/src/BarChart/story/index.jsx:117]: no, only to remove the `delay: 500` line 😃

File: packages/picasso-charts/src/BarChart/BarChart.tsx (hunk 9 above)
Comment (DmitriyT51) [reply packages/picasso-charts/src/BarChart/BarChart.tsx]: agree, I refactored naming

File: packages/picasso-charts/src/BarChart/story/index.jsx (hunk 10 above)
Comment (DmitriyT51) [reply packages/picasso-charts/src/BarChart/story/index.jsx]: done ;)

File: cypress/component/BarChart.spec.tsx (hunk 8 above)
Comment (denieler) [reply cypress/component/BarChart.spec.tsx:155]: I'm preparing a fix for this, but I probably would make it in a separate PR after this one will be merged. Just to make code review process easier 🙇

**Review by denieler — APPROVED** [review]

**Pull request merged**